.github/workflows/unittest.yaml

```diff
@@ -32,5 +35,135 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
 
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup pixi
+        uses: prefix-dev/setup-pixi@v0.8.10
+        with:
+          pixi-version: v0.41.4
+          manifest-path: pyproject.toml
+
       - name: Build conda package
         run: pixi run conda-build
+
+      - name: Upload conda package as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifact-conda-package
+          path: ${{ env.PKG_NAME }}-*.conda
+
+  conda-verify:
+    needs: build
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Setup micromamba
+        uses: mamba-org/setup-micromamba@v2
+        with:
+          environment-name: test
+          init-shell: bash
+          create-args: >-
+            python=3.10
+
+      - name: Download conda package artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: artifact-conda-package
+          path: /tmp/local-channel/linux-64
+
+      - name: Install the package
+        run: |
+          micromamba install --yes -c conda-forge conda-build conda-index
+          python -m conda_index /tmp/local-channel
+          micromamba install --yes -c /tmp/local-channel -c mantid-ornl -c conda-forge ${{ env.PKG_NAME }}
+
+      - name: Verify the installation
+        run: |
+          conda_version=$(micromamba list "${{ env.PKG_NAME }}" | awk -v pkg="${{ env.PKG_NAME }}" '$1 == pkg { print $2 }')
+          echo "Conda version: $conda_version"
+          wheel_version=$(python -c "import ${{ env.PKG_NAME }}; print(${{ env.PKG_NAME }}.__version__)")
+          echo "Wheel version: $wheel_version"
+          if [ "$conda_version" != "$wheel_version" ]; then
+            echo "Version mismatch!"
+            exit 1
+          else
+            echo "Versions match."
+          fi
```

Comment:
again, it seems like the entire workflow is centered around micromamba.
Given the heavy reliance on micromamba, I would recommend making a separate micromamba focused feature branch instead of trying to mix using both pixi and micromamba.
This will probably make a lot of software packaging work easier especially when existing software build system is heavily invested with micromamba and conda suite.